### ISSUE TITLE
[SwiftUI] After downloading image the SwiftUI view doesn't trigger a redraw

### ISSUE THREAD

**Issue by hyouuu** [onevcat/Kingfisher#1385]
### Check List

- [x] I have read the [wiki page](https://github.com/onevcat/Kingfisher/wiki) and [cheat sheet](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet), but there is no information I need.
- [x] I have searched in [existing issues](https://github.com/onevcat/Kingfisher/issues?utf8=✓&q=is%3Aissue), but did not find a same one.
- [x] I want to report a problem instead of asking a question. It'd better to use [kingfisher tag in Stack Overflow](http://stackoverflow.com/questions/tagged/kingfisher) to ask a question.

### Issue Description
After downloading image from an URL, the KFImage wouldn't trigger a SwiftUI redraw, so that the image view is still blank; if I change any @State var in the SwiftUI view, the image appears.

#### Reproduce

Better described with the usage I have - 

```swift
class User {
    var imageKey = "abc"    
    
    func kfImageSource() -> Kingfisher.Source? {
        guard let url = imageDownloadURL() else { return nil }
        let imageResource = ImageResource(downloadURL: url, cacheKey: imageKey)
        return Kingfisher.Source.network(imageResource)
    }

    func imageDownloadURL() -> URL? {
        guard let imageKey = imageKey else { return nil }
        return AWSRequer.preSignedURL(for: imageKey) 
    }
}

struct UserRow: View {
    @State var someNum = 1
    var user: User

    var body: some View {
            KFImage(source: user.kfImageSource())
                .placeholder {
                    // Placeholder while downloading.
                    Image(systemName: "arrow.2.circlepath.circle")
                        .font(.largeTitle)
                        .opacity(0.3)
                }
            Text("\(someNum)")
    }
}
```
Basically this will always show the placeholder, and if I set a 30 seconds timer or something to change `someNum` to 2, then the actual KFImage appears.

**Comment by onevcat** [comment]
Can you make sure that the first time you get a correct non-nil URL? I also suggest you can add two handler to see if either result is returned by Kingfisher:

```swift
KFImage(source: user.kfImageSource())
  .onSuccess { r in
    print("suc: \(r)")
  }
  .onFailure { e in
    print("err: \(e)")
  }
```

**Comment by hyouuu** [comment]
Good catch! It turns out AWSS3 preSignedURL sometimes return asynchronously so the URL could be nil. 
Now the new question is - could we setup an `ImageResource` which takes a completion block for URL generation then start downloading the image?

**Comment by onevcat** [comment]
😂 No, not yet. Kingfisher does not support it out-of-box. But you can always create a simple wrapper to trigger the `@State` change in the URL. (Or make some modifications to Kingfisher's code and send a PR...)

**Comment by hyouuu** [comment]
Thanks and totally understand! Will see if could get time to add this functionality to KF sometime :)
